src/main/kotlin/ai/digital/integration/server/deploy/internals/cluster/operator/AwsEksHelper.kt

```diff
@@ -1,24 +1,391 @@
+                    logOutput = true,
+                    throwErrorOnFailure = false)
+
+        }
+    }
+
+    private fun existingSshKeyPair(): Boolean {
+        val result: String = ProcessUtil.executeCommand(project,
+                "aws  --region ${getProvider().region.get()} " +
+                        "ec2 describe-key-pairs " +
+                        "--key-names=${getProvider().sshKeyName.get()} " +
+                        "--query 'KeyPairs[0].KeyName' " +
+                        "--output text",
+                logOutput = true,
+                throwErrorOnFailure = false)
+        if (result.contains("does not exist")) {
+            return false
+        } else if (result.contains(getProvider().sshKeyName.get())) {
+            return true
+        }
+        return false
+    }
+
+    private fun createCluster(skipExisting: Boolean) {
+        val shouldSkipExisting = if (skipExisting) {
+            existingStack()
+        } else {
+            false
+        }
+        if (shouldSkipExisting) {
+            project.logger.lifecycle("Skipping creation of the stack: {}", getProvider().stack.get())
+        } else {
+            project.logger.lifecycle("Create Stack  : {}", getProvider().stack.get())
+            val awsEksClusterTemplateFile = getTemplate("operator/aws-eks/aws-eks-cluster.yaml")
+            val awsEksProvider: AwsEksProvider = getProvider()
+
+            val stackId = ProcessUtil.executeCommand(project,
+                    "aws --region  ${awsEksProvider.region.get()} " +
+                            "cloudformation create-stack " +
+                            "--stack-name ${awsEksProvider.stack.get()} " +
+                            "--template-body file://$awsEksClusterTemplateFile " +
+                            "--capabilities CAPABILITY_IAM " +
+                            "--parameters " +
+                            "ParameterKey=ProjectName,ParameterValue=${awsEksProvider.stack.get()} " +
+                            "ParameterKey=ClusterName,ParameterValue=${awsEksProvider.clusterName.get()} " +
+                            "ParameterKey=NodeGroupName,ParameterValue=${awsEksProvider.nodeGroupName.get()} " +
+                            "ParameterKey=KeyName,ParameterValue=${awsEksProvider.sshKeyName.get()} " +
+                            "ParameterKey=FileSystemName,ParameterValue=${awsEksProvider.fileSystemName.get()} " +
+                            "ParameterKey=NodeDesiredSize,ParameterValue=${awsEksProvider.clusterNodeCount.get()} " +
+                            "ParameterKey=KubernetesVersion,ParameterValue=${awsEksProvider.kubernetesVersion.get()} " +
+                            "--on-failure DELETE " +
+                            "--output text",
+                    logOutput = true)
+            verifyClusterStatus(1500000, 300000, stackId, "CREATE_COMPLETE")// delay time 25 mins and sleep time 5 mins
```

Comment:
Please move all timeouts on configurable level, and please remove all comments related to it

Reply:
done